packages/boutique/frontend/vite.config.ts

```diff
@@ -8,6 +8,9 @@ export default defineConfig({
     port: 4004
   },
   resolve: {
-    alias: [{ find: '@', replacement: resolve(__dirname, 'src') }]
+    alias: {
+      '@': resolve(__dirname, 'src'),
+      '@shared/boutique': resolve(__dirname, '../../boutique/shared/src')
```

Comment:
No need to have the type alias if `@shared/boutique` is a package.

Reply:
well without this we will get  `Failed to resolve entry for package "@boutique/shared"` error in locally run build by docker

Reply:
I've looked into this. The main reason behind this error is that `vite` is expecting the source code to be ESM only. and we are bundling `@boutique/shared` as CJS. This should be okay for now, but I think we should discuss if we want to move everything to ESM. A lot of libraries are starting to be bundled for ESM only.
- https://github.com/wooorm/npm-esm-vs-cjs?tab=readme-ov-file#data

Node 22 will have support for importing ESM into CJS using `require`, but we will be able to use this functionality in Octomber/November.
- https://nodejs.org/en/blog/announcements/v22-release-announce#support-requireing-synchronous-esm-graphs
- https://nodejs.org/en/about/previous-releases

Converting the shared package to ESM will throw a runtime error when boutique's backend is ran:
```
Error [ERR_REQUIRE_ESM]: require() of ES Module /Users/radu/dev/testnet/packages/boutique/shared/dist/index.js from /Users/radu/dev/testnet/packages/boutique/backend/dist/order/controller.js not supported.
Instead change the require of index.js in /Users/radu/dev/testnet/packages/boutique/backend/dist/order/controller.js to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/Users/radu/dev/testnet/packages/boutique/backend/dist/order/controller.js:7:18) {
  code: 'ERR_REQUIRE_ESM'
}
```